hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/server/namenode/ha/ObserverReadProxyProvider.java

```diff
@@ -368,6 +368,21 @@ private boolean shouldFindObserver() {
     return true;
   }
 
+  private void confirmActiveToMsync() throws IOException {
+    ClientProtocol activeProxy = (ClientProtocol) getCurrentProxy().proxy;
+    for (int i = 0; i < nameNodeProxies.size(); i++) {
+      NNProxyInfo<T> current = getCurrentProxy();
+      HAServiceState currState = current.getCachedState();
+      if (currState == HAServiceState.ACTIVE) {
+        activeProxy = getProxyAsClientProtocol(current.proxy);
+        LOG.info("active proxy is choosed :{}", current);
+        break;
+      } else {
+        changeProxy(current);
```

Comment:
hmm, the current proxy in observer proxy provider in general should be pointing to Observer namenode, you seems to change  the current proxy itself to active here? What would happen to subsequent calls, will they go to active or the concurrent calls

Reply:
> 

Hi, @ayushtkn . the subsequent calls will not go to active.